client-ui/src/app/components/time-frame-picker/time-frame-picker.component.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import * as moment from 'moment';
+import {NgbDateStruct, NgbTimeStruct} from "@ng-bootstrap/ng-bootstrap";
+import {DateRange, Timestamp} from "./time-frame-picker.model";
+
+type ViewModel = [LocalDateTimeModel, LocalDateTimeModel]
+
+interface LocalDateTimeModel {
+  time: NgbTimeStructWithMillis
+  date: NgbDateStruct
+}
+
+interface NgbTimeStructWithMillis extends NgbTimeStruct {
+  millis: number
+}
+
+@Component({
+  selector: 'time-frame-picker',
+  templateUrl: './time-frame-picker.component.html',
+  styleUrls: ['./time-frame-picker.component.less']
+})
+export class TimeFramePickerComponent {
+  model: ViewModel
+  modelBoundaries: ViewModel
+
+  @Input()
+  public set range(range: DateRange) {
+    console.log("@Input() range", range)
+    this.model = <ViewModel>range.map(TimeFramePickerComponent.timestampToLdt)
+  }
+
+  @Input()
+  public set rangeBoundaries(boundaries: DateRange | undefined) {
+    console.log("@Input() boundaries", boundaries)
+    this.modelBoundaries = boundaries
+      ? <ViewModel>boundaries.map(TimeFramePickerComponent.timestampToLdt)
+      : this.model
+  }
+
+  @Output()
+  public rangeChange = new EventEmitter<DateRange>()
+
+  onTimeFromChange(time: NgbTimeStruct) {
+    const [from, till] = this.model
+    this.onModelChange([{...from, time: {...time, second: 0, millis: 0}}, till])
+  }
+
+  onTimeTillChange(time: NgbTimeStruct) {
+    const [from, till] = this.model
+    this.onModelChange([from, {...till, time: {...time, second: 59, millis: 999}}])
+  }
+
+  onSelectedDatesChange([dateFrom, dateTill]: NgbDateStruct[]) {
+    this.onModelChange([
+      {date: dateFrom, time: {hour: 0, minute: 0, second: 0, millis: 0}},
+      {date: dateTill, time: {hour: 23, minute: 59, second: 59, millis: 999}}
+    ])
+  }
+
+  private onModelChange(model: ViewModel): void {
+    console.log("@Output() range", model.map(TimeFramePickerComponent.ldtToTimestamp))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Calls to 'console.log' are not allowed.](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4491035)